src/components/application_manager/src/commands/hmi/on_driver_distraction_notification.cc

```diff
@@ -51,38 +99,31 @@ OnDriverDistractionNotification::~OnDriverDistractionNotification() {}
 
 void OnDriverDistractionNotification::Run() {
   LOG4CXX_AUTO_TRACE(logger_);
-
   const hmi_apis::Common_DriverDistractionState::eType state =
       static_cast<hmi_apis::Common_DriverDistractionState::eType>(
           (*message_)[strings::msg_params][hmi_notification::state].asInt());
-  application_manager_.set_driver_distraction(state);
+  application_manager_.set_distracting_driver_state(state);
 
   smart_objects::SmartObjectSPtr on_driver_distraction =
-      new smart_objects::SmartObject();
+      utils::MakeShared<smart_objects::SmartObject>();
 
   if (!on_driver_distraction) {
     LOG4CXX_ERROR(logger_, "NULL pointer");
     return;
   }
-
   (*on_driver_distraction)[strings::params][strings::function_id] =
       mobile_api::FunctionID::OnDriverDistractionID;
-
+  (*on_driver_distraction)[strings::params][strings::message_type] =
+      static_cast<int32_t>(application_manager::MessageType::kNotification);
   (*on_driver_distraction)[strings::msg_params][mobile_notification::state] =
       state;
 
   const ApplicationSet applications =
       application_manager_.applications().GetData();
 
-  ApplicationSetConstIt it = applications.begin();
-  for (; applications.end() != it; ++it) {
-    const ApplicationSharedPtr app = *it;
-    if (app) {
-      (*on_driver_distraction)[strings::params][strings::connection_key] =
-          app->app_id();
-      SendNotificationToMobile(on_driver_distraction);
-    }
-  }
+  ApplicationProcessor app_processor(
```

Comment:
@LitvinenkoIra please rename ApplicationProcessor class to a better name. It should show that it process driver destruction messages